### DISCUSSION TITLE
Including a file, expanding the macros in it and writing the result to another file?

### DISCUSSION THREAD

**Discussion by daandemeyer** [rpm-software-management/rpm#2912] [Q&A]
I'm trying to include a file with macros in it, expand those macros and write the result to another file. My last attempt that doesn't work is the following:

```
echo "%{expand:%include %{SOURCE203}}" > expanded
```

If SOURCE203 contains the following:

```
%{_unitdir}/systemd-coredump.socket
%{_unitdir}/systemd-coredump@.service
```

I want "expanded" to contain:

```
/usr/lib/systemd/system/systemd-coredump.socket
/usr/lib/systemd/system/systemd-coredump@.service
```

All rpm macros defined in the spec file should be taken into account so shelling out to `rpm --eval` won't work I think.

Anyone able to help me out?

**Comment by Conan-Kudo** [comment]
This should do the trick:

```
cat > expanded <<EOF
%{expand:%include %{SOURCE203}}
EOF
```
